cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -146,11 +146,21 @@ func (c *Client) Sync() error {
 	}
 	defer cli.Close()
 
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	err = cli.Sync(ctx)
-	cancel()
+	// Syncs the list of endpoints
+	var lastError error
+	err = wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
The combination of `ExponentialBackoff` with the current settings and the timeout context with 5 sec timeout is going to cause some severe delays (each try has a 5 sec timeout, followed by an exponentially increasing sleep).
Therefore I am not sure if `ExponentialBackoff` needs to be used here at all.

Reply:
as per slack discussion, timeout trimmed down to 2 sec